### DISCUSSION TITLE
what is the function of account type?

### DISCUSSION THREAD

**Discussion by laptopmutia** [frappe/books#380] [Q&A] [answered]
I'm trying to learn the codebase and the accountings part

anyone could explain to me what is the function of account type?

**Comment by 18alantom** [comment, marked as answer]
In Books, account type is a parameter that describes the purpose of that account and to some extent, its place in the [Chart of Accounts](https://en.wikipedia.org/wiki/Chart_of_accounts), it's also used for the selection and listing of a default account for some purpose.

For instance, when making a _Payment_ entry for receiving a cash payment, Books lists out accounts that have their account type set to _Cash_.

**Reply by laptopmutia** [reply]
did it means it should hard locked to the root type?

I mean if a account type is cash it should be asset right?

also in frappe the account type are only 
```
'Accumulated Depreciation',
        'Bank',
        'Cash',
        'Chargeable',
        'Cost of Goods Sold',
        'Depreciation',
        'Equity',
        'Expense Account',
        'Expenses Included In Valuation',
        'Fixed Asset',
        'Income Account',
        'Payable',
        'Receivable',
        'Round Off',
        'Stock',
        'Stock Adjustment',
        'Stock Received But Not Billed',
        'Tax',
        'Temporary',
    ```
    
    but in practice we could add or remove it as needed right?

**Reply by 18alantom** [reply]
Yup, if the account type is _Cash_ it will have the root type of _Asset_

Here's how the typing will look for this

```typescript
export type AccountType =
  | 'Accumulated Depreciation'
  | 'Bank'
  | 'Cash'
  | 'Chargeable'
  | 'Cost of Goods Sold'
  | 'Depreciation'
  | 'Equity'
  | 'Expense Account'
  | 'Expenses Included In Valuation'
  | 'Fixed Asset'
  | 'Income Account'
  | 'Payable'
  | 'Receivable'
  | 'Round Off'
  | 'Stock'
  | 'Stock Adjustment'
  | 'Stock Received But Not Billed'
  | 'Tax'
  | 'Temporary';

export type AccountRootType =
  | 'Asset'
  | 'Liability'
  | 'Equity'
  | 'Income'
  | 'Expense';

```

Yup, you could add new ones, as of now in Books this is not supported, so when creating a new account one of the above `AccountType`s has to be selected.
